lib/common/json-schema-validator.js

```diff
@@ -90,25 +95,105 @@ function jsonSchemaValidatorFactory(
     };
 
     /**
-     * Get compiled schema from the instance by `key` or `ref`.
-     * @param  {String} key `key` that was passed to `addSchema` or
-     *                  full schema reference (`schema.id` or resolved id).
+     * Add schema to the instance.
+     * @param {Object} schema
+     * @param {String} name Optional schema name when schema has an id
+     */
+    JsonSchemaValidator.prototype.addSchema = function (schema, name) {
+        assert.equal(_.isEmpty(name) && _.isEmpty(schema.id), false,
+                '`name` should not be empty, or `schema` should has an `id` property');
+
+        schema.id = url.resolve(this.nameSpace, name || schema.id);
+        return this._ajv.addSchema(schema);
+    };
+
+    /**
+     * Add array of schemas to the instance.
+     * @param {Array} schemas array of schema
+     */
+    JsonSchemaValidator.prototype.addSchemas = function (schemas) {
+        assert.arrayOfObject(schemas,
+                '`schemas` should be an array of JSON schema');
+        var self = this;
+        _.forEach(schemas, function (schema) {
+            self.addSchema(schema);
+        });
+    };
+
+    /**
+     * Add all schema in the folder to the instance.
+     * @param {String} schemaDir the directory where schema files in
+     * @param {String} metaSchemaName Optional the meta schema name
+     */
+    JsonSchemaValidator.prototype.addSchemasByDir = function (schemaDir, metaSchemaName) {
+        var self = this;
+
+        assert.string(schemaDir, 'schemaDir');
+        if (metaSchemaName) {
+            assert.string(metaSchemaName, 'metaSchemaName');
+        }
+
+        var fileNames = _.filter(fs.readdirSync(schemaDir), function (fileName) {
+            return path.extname(fileName).toLowerCase() === '.json';
+        });
+
+        var schemas = [];
+        var metaSchema;
+        _.forEach(fileNames, function(fileName) {
+            var fullName = path.resolve(schemaDir, fileName);
+            var data = fs.readFileSync(fullName);
+            var name = path.basename(fileName.toLowerCase(), '.json');
+            var content = JSON.parse(data.toString());
+            content.id = name;
+            if (metaSchemaName === fileName) {
+                metaSchema = content;
+            } else {
+                schemas.push(content);
+            }
+        });
+
+        //meta schema should be added first, so subsequent schemas can be validated againt it
+        if (metaSchemaName) {
+            if (!metaSchema) {
+                throw Errors.NotFoundError('Cannot find the meta schema "' + metaSchemaName +
+                    '" in directory "' + schemaDir + '"');
+            }
+            self.addMetaSchema(metaSchema);
```

Comment:
I think the logic here is correct.
1. make sure `metaSchemaName` param passed and is not empty
2. if no `metaSchema` value found , throw error. 
3. otherwise , `addMetaSchema`